src/ring_theory/polynomial/gauss_lemma.lean

```diff
@@ -43,34 +49,129 @@ begin
+  { intros a ha, obtain ⟨n, -, rfl⟩ := mem_frange_iff.1 ha,
+    obtain ⟨p, hp, he⟩ := this n, use [p, hp],
+    rw [is_scalar_tower.algebra_map_eq R K, coeff_map, ← eval₂_map, eval₂_at_apply] at he,
+    rw eval₂_eq_eval_map, apply (injective_iff_map_eq_zero _).1 _ _ he,
+    { apply ring_hom.injective } },
+  rw [is_scalar_tower.algebra_map_eq R K _, ← map_map],
+  refine multiset.mem_of_le (roots.le_of_dvd ((hf.map _).map _).ne_zero _) ha,
+  { apply_instance },
+  { exact map_dvd (algebra_map K g.splitting_field) hd },
+  { apply splitting_field_aux.is_scalar_tower },
+end
+
+variables [is_fraction_ring R K]
+
+theorem eq_map_of_dvd [is_integrally_closed R] {f : R[X]} (hf : f.monic)
+  (g : K[X]) (hg : g.monic) (hd : g ∣ f.map (algebra_map R K)) :
+  ∃ g' : R[X], g'.map (algebra_map R K) = g :=
+begin
+  let algeq := (subalgebra.equiv_of_eq _ _ $
+    is_integrally_closed.integral_closure_eq_bot R _).trans
+    (algebra.bot_equiv_of_injective $ is_fraction_ring.injective R $ K),
+  have : (algebra_map R _).comp algeq.to_alg_hom.to_ring_hom =
+    (integral_closure R _).to_subring.subtype,
+  { ext, conv_rhs { rw ← algeq.symm_apply_apply x }, refl },
+  refine ⟨map algeq.to_alg_hom.to_ring_hom _, _⟩,
+  use g.to_subring _ (frange_subset_integral_closure hf hg hd),
+  rw [map_map, this],
+  apply g.map_to_subring,
+end
+
+lemma temporary [is_integrally_closed R] {f : R[X]} (hf : f.monic)
+  (g : K[X]) (hd : g ∣ f.map (algebra_map R K)) :
+  ∃ g' : R[X], (g'.map (algebra_map R K)) * (C $ leading_coeff g) = g :=
+begin
+  sorry
+end
 
 lemma is_primitive.is_unit_iff_is_unit_map {p : R[X]} (hp : p.is_primitive) :
   is_unit p ↔ is_unit (p.map (algebra_map R K)) :=
 hp.is_unit_iff_is_unit_map_of_injective (is_fraction_ring.injective _ _)
 
+lemma monic_eq_prod_coeff_inv {a b c : R[X]} (ha : a.monic) (h : a = b * c) :
+  leading_coeff b * leading_coeff c = 1 := sorry
+
+theorem monic.irreducible_iff_irreducible_map_fraction_map [is_integrally_closed R]
+  {p : R[X]} (h : p.monic) :
+  irreducible p ↔ irreducible (p.map $ algebra_map R K) :=
+begin
+  refine ⟨λ hp, _, λ hp, _⟩,
+  rw irreducible_iff,
+  split,
+    sorry,
+  intros a b H,
+
+  rw or_iff_not_imp_left,
+  intro ha,
+  --rw is_primitive.is_unit_iff_is_unit_map K,
+  obtain ⟨a', ha⟩ := temporary h a (dvd_of_mul_right_eq _ H.symm),
+  obtain ⟨b', hb⟩ := temporary h b (dvd_of_mul_left_eq _ H.symm),
+  have := monic_eq_prod_coeff_inv sorry H,
+  rw [← ha, ← hb, mul_comm _ (C b.leading_coeff), mul_assoc, ← mul_assoc (C a.leading_coeff),
+    ← C_mul, this, C_1, one_mul, ← polynomial.map_mul] at H,
+  have := polynomial.map_injective _ (is_fraction_ring.injective R K) H,
+  have other := hp.is_unit_or_is_unit this,
+
+  --obtain ⟨a', ha⟩ := eq_map_of_dvd h
+  sorry,
+  exact is_primitive.irreducible_of_irreducible_map_of_injective (is_fraction_ring.injective R K)
+    h.is_primitive hp,
+end
+
 open is_localization
 
+variable [normalized_gcd_monoid R]
```

Comment:
You are in a section called `normalized_gcd_monoid` well before having this assumption, you can maybe reorganize a little bit the file.